src/ppx/geometry.cpp

```diff
@@ -19,6 +19,334 @@
+        }
+        uint32_t endSize = GetVertexBufferSize(pGeom, kBufferIndex);
+
+        uint32_t bytesWritten = (endSize - startSize);
+        PPX_ASSERT_MSG((bytesWritten == GetVertexBufferElementSize(pGeom, kBufferIndex)), "size of vertex data written does not match buffer's element size");
+
+        return GetVertexBufferElementCount(pGeom, kBufferIndex);
+    }
+    virtual uint32_t AppendVertexData(Geometry* pGeom, const WireMeshVertexData& vtx) override
+    {
+        uint32_t       startSize = GetVertexBufferSize(pGeom, kBufferIndex);
+        const uint32_t attrCount = GetVertexBindingAttributeCount(pGeom, kBufferIndex);
+        for (uint32_t attrIndex = 0; attrIndex < attrCount; ++attrIndex) {
+            const grfx::VertexSemantic semantic = GetVertexBindingAttributeSematic(pGeom, kBufferIndex, attrIndex);
+
+            // clang-format off
+        switch (semantic) {
+            default: break;
+            case grfx::VERTEX_SEMANTIC_POSITION  : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.position); break;
+            case grfx::VERTEX_SEMANTIC_COLOR     : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.color); break;
+        }
+            // clang-format on
+        }
+        uint32_t endSize = GetVertexBufferSize(pGeom, kBufferIndex);
+
+        uint32_t bytesWritten = (endSize - startSize);
+        PPX_ASSERT_MSG(bytesWritten == GetVertexBufferElementSize(pGeom, kBufferIndex), "size of vertex data written does not match buffer's element size");
+
+        return GetVertexBufferElementCount(pGeom, kBufferIndex);
+    }
+    virtual uint32_t GetVertexCount(const Geometry* pGeom) override
+    {
+        return GetVertexBufferElementCount(pGeom, kBufferIndex);
+    }
+
+private:
+    const uint32_t kBufferIndex = 0;
+};
+
+class VertexDataProcessorPositionPlanar : public IVertexDataProcessor
+{
+public:
+    virtual bool Validate(Geometry* pGeom) override
+    {
+        return true;
+    }
+    virtual Result UpdateVertexBuffer(Geometry* pGeom) override
+    {
+        PPX_ASSERT_MSG((2 == GetVertexBindingCount(pGeom)), "there should be 2 binding for position planar");
+        // Position
+        AddVertexBuffer(pGeom, kPositionBufferIndex);
+        // Non-Position data
+        AddVertexBuffer(pGeom, kNonPositionBufferIndex);
+
+        SetPositionBufferIndex(pGeom, kPositionBufferIndex);
+
+        const uint32_t attributeCount = GetVertexBindingAttributeCount(pGeom, kNonPositionBufferIndex);
+        for (uint32_t i = 0; i < attributeCount; ++i) {
+            const grfx::VertexSemantic semantic = GetVertexBindingAttributeSematic(pGeom, kNonPositionBufferIndex, i);
+            // clang-format off
+            switch (semantic) {
+                default: {
+                    return Result::ERROR_GEOMETRY_INVALID_VERTEX_SEMANTIC;
+                }
+
+                case grfx::VERTEX_SEMANTIC_POSITION  : 
+                    PPX_ASSERT_MSG(false, "position should be in binding 0"); break;
```

Comment:
This should be on the same line as the case statement if the switch block is surrounded by clang-format tags.

Reply:
fixed